### PR TITLE
feat: add shutdown

### PR DESCRIPTION
**Pull request by Hugome**
# Description

Add Shutdown action

## Issue reference

Related to dapr/dapr#2828

## Checklist

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Comment by wcs1only** [comment]
Unit tests are failing, probably just need a shutdown implementation in tests/clients/fake_dapr_server.py ?

**Comment by wcs1only** [comment]
> Unit tests are failing, probably just need a shutdown implementation in tests/clients/fake_dapr_server.py ?

I sent you a PR @Hugome with a patch that makes the unit tests pass:
https://github.com/Hugome/python-sdk/pull/1

**Comment by Hugome** [comment]
Merged, thanks for the fix :)

File: dapr/clients/grpc/client.py [hunk 1]
```diff
@@ -809,3 +810,22 @@ def wait(self, timeout_s: float):
...
+
+        This will ask the sidecar to gracefully shutdown.
+
+        The example shutdown the sidecar:
+
+            from dapr.clients import DaprClient
+
+            with DaprClient() as d:
+                resp = d.shutdown()
+        """
```
Comment (wcs1only) [inline dapr/clients/grpc/client.py:827]:
```suggestion

        Args:
            metadata (tuple, optional): custom grpc metadata
        """
```

Though I'm not sure it is strictly necessary to take metadata as a parameter at all here. We don't do anything with it on the runtime side, correct?

**Comment by wcs1only** [comment]
Marking this do-not-merge pending dapr/dapr#2828. If that makes it into the v1.1.0 release, we can safely merge this PR. Otherwise, we'll wait until after v1.1.0 release.

File: dapr/clients/grpc/client.py (hunk 1 above)
Comment (Hugome) [reply dapr/clients/grpc/client.py:827]: I was not sure for the metadata, does the auth token (https://docs.dapr.io/operations/security/api-token/) can go through those metadata ? Or it's managed by the `DaprClient` ?

Comment (wcs1only) [reply dapr/clients/grpc/client.py:827]: We have an interceptor add the token to every request: 
https://github.com/dapr/python-sdk/blob/master/dapr/clients/grpc/client.py#L89-L93

The user shouldn't need to pass anything themselves.

**Review by wcs1only — APPROVED** [review]

**Pull request merged**